WalletWasabi.Backend/Config.cs

```diff
@@ -27,59 +27,39 @@ public class Config : IConfig
 		[JsonProperty(PropertyName = "BitcoinRpcConnectionString")]
 		public string BitcoinRpcConnectionString { get; private set; }
 
-		[JsonProperty(PropertyName = "MainNetBitcoinCoreHost")]
-		public string MainNetBitcoinCoreHost { get; internal set; }
+		[JsonProperty(PropertyName = "MainNetBitcoinP2pEndPoint")]
+		[JsonConverter(typeof(EndPointJsonConverter), Constants.DefaultMainNetBintcoinP2pPort)]
+		public EndPoint MainNetBitcoinP2pEndPoint { get; internal set; }
 
-		[JsonProperty(PropertyName = "TestNetBitcoinCoreHost")]
-		public string TestNetBitcoinCoreHost { get; internal set; }
+		[JsonProperty(PropertyName = "TestNetBitcoinP2pEndPoint")]
+		[JsonConverter(typeof(EndPointJsonConverter), Constants.DefaultTestNetBintcoinP2pPort)]
+		public EndPoint TestNetBitcoinP2pEndPoint { get; internal set; }
 
-		[JsonProperty(PropertyName = "RegTestBitcoinCoreHost")]
-		public string RegTestBitcoinCoreHost { get; internal set; }
-
-		[JsonProperty(PropertyName = "MainNetBitcoinCorePort")]
-		public int? MainNetBitcoinCorePort { get; internal set; }
-
-		[JsonProperty(PropertyName = "TestNetBitcoinCorePort")]
-		public int? TestNetBitcoinCorePort { get; internal set; }
-
-		[JsonProperty(PropertyName = "RegTestBitcoinCorePort")]
-		public int? RegTestBitcoinCorePort { get; internal set; }
+		[JsonProperty(PropertyName = "RegTestBitcoinP2pEndPoint")]
+		[JsonConverter(typeof(EndPointJsonConverter), Constants.DefaultRegTestBintcoinP2pPort)]
+		public EndPoint RegTestBitcoinP2pEndPoint { get; internal set; }
 
 		private EndPoint _bitcoinCoreEndPoint;
 
-		public EndPoint GetBitcoinCoreEndPoint()
+		public EndPoint GetBitcoinP2pEndPoint()
 		{
 			if (_bitcoinCoreEndPoint is null)
```

Comment:
I know this was not introduced in this PR but this makes no sense at all.

Reply:
Bitcoin P2P is a Bitcoin protocol, not Bitcoin Core specific. RPC is Bitcoin Core specific protocol. Thus `BitcoinP2p` and `BitcoinCoreRpc`.

What may confuses you that I forgot to rename the backing field. Done: https://github.com/zkSNACKs/WalletWasabi/pull/1945/commits/d3b2f3e2fe9b860ef4c1b04100944e3932507d0c

Reply:
No no, what I say makes no sense is the:
```c#
if (_bitcoinCoreEndPoint is null)
```

Reply:
Indeed: https://github.com/zkSNACKs/WalletWasabi/pull/1945/commits/9558f1c04995ab528485cc34954a0059617d0578